pkg/controller/deadmanssnitch/deadmanssnitch_controller.go

```diff
@@ -244,21 +275,17 @@ func newSyncSet(namespace string, clusterDeploymentName string, snitchURL string
 			},
 			SyncSetCommonSpec: hivev1alpha1.SyncSetCommonSpec{
 				ResourceApplyMode: "sync",
-				Resources: []runtime.RawExtension{
+				// Use SecretReference here which comsume the secret in the cluster namespace,
+				// instead of embed the secret in the SyncSet directly
+				SecretReferences: []hivev1alpha1.SecretReference{
 					{
-						Object: &corev1.Secret{
-							Type: "Opaque",
-							TypeMeta: metav1.TypeMeta{
-								Kind:       "Secret",
-								APIVersion: "v1",
-							},
-							ObjectMeta: metav1.ObjectMeta{
-								Name:      "dms-secret",
-								Namespace: "openshift-monitoring",
-							},
-							Data: map[string][]byte{
-								config.KeySnitchURL: []byte(snitchURL),
-							},
+						Source: corev1.ObjectReference{
+							Name:      clusterDeploymentName + config.SecretPostfix,
```

Comment:
Recommend passing in the source secret name as a function arg rather than reconstructing.

Reply:
moved to function arg